### PR TITLE
Support more runtimes, simplify public interface

### PR DESCRIPTION
**Pull request by humphd**
This adds support for more runtimes (deno, bun, node.js, browser, etc) and simplifies the public interface to be a single function, `decryptSops()`, which can accept a file-like object/stream, or an options Object with filepath or URL where a SOPS file can be found.

I've cleaned up some old code/files that I missed before.

I should add tests for Deno, Bun, Browser, etc. but it can happen in a follow-up.

### CODE REVIEW

**Comment by humphd** [comment]
In a follow-up, the rest of the node-specific stuff should get removed (e.g., `process.env.*`).

**Comment by humphd** [comment]
Added smoke tests for node (CJS), deno (ESM/TS), and bun (ESM/TS).  Made env var access work in deno and bun as well as node. Moved test related stuff below `tests/*`

File: tests/smoke/deno/deno.test.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+// @ts-ignore - need to build before this will exist
 import { decryptSops } from "../../../dist/index.js";
```
Comment (tarasglek) [inline tests/smoke/deno/deno.test.ts:2]: not sure about bun, but for deno the correct solution is to import-map sops-age to ../../../src/index.ts

**Review by tarasglek — APPROVED** [review]
This is really impressive. I didn't think about adding a smoketest for deno/bun. Would be nice to smoketest browser too, but no idea how to do that.

File: tests/smoke/deno/deno.test.ts (hunk 1 above)
Comment (tarasglek) [reply tests/smoke/deno/deno.test.ts:2]: we can leave this as is, more for future reference for experimentation

Comment (humphd) [reply tests/smoke/deno/deno.test.ts:2]: For smoke tests, I wanted to use the built version vs. what's in `src/` so I can test how the various formats get bundled.  We could add a `prepare` script to build things when you install, or even put `dist` in git; but I think the above is fine for testing.

**Comment by tarasglek** [comment]
> For browser, we can use Playwright, but I didn't feel like (re)learning it over the holidays. I'll add one when I have some time.

i was hoping there was some playwright vitest mode by now.

**Comment by humphd** [comment]
Playwright basically is Vitest, it won't be hard to add.

**Pull request merged**